exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/FunctionImplementationRegistry.java

```diff
@@ -186,4 +226,105 @@ public boolean isFunctionComplexOutput(String name) {
     return false;
   }
 
+  public RemoteFunctionRegistry getRemoteFunctionRegistry() {
+    return remoteFunctionRegistry;
+  }
+
+  public List<Func> validate(Path path) throws IOException {
+    URL url = path.toUri().toURL();
+    URL[] urls = {url};
+    ClassLoader classLoader = new URLClassLoader(urls);
+    return drillFuncRegistry.validate(path.getName(), scan(classLoader, path, urls));
+  }
+
+  public void register(String jarName, ScanResult classpathScan, ClassLoader classLoader) {
+    drillFuncRegistry.register(jarName, classpathScan, classLoader);
+  }
+
+  public void unregister(String jarName) {
+    drillFuncRegistry.unregister(jarName);
+  }
+
+  /**
+   * Loads all missing functions from remote registry.
+   * Compares list of already registered jars and remote jars, loads missing jars.
+   * Missing jars are stores in local DRILL_UDF_DIR.
```

Comment:
Yes, I'll add note that it's an environmental variable but after chnages it will be $DRILL_CONF_DIR/udf.